### ISSUE TITLE
The Middleware of header can not modify Host header

### ISSUE THREAD

**Issue by sifksky** [containous/traefik#6000]
Hi Traefik:
    I read all traefik doc from headers and I test all parameter from headers. I still can not modify the Host header. But I can modify the header "X-Forwarded-Host" from customRequestHeaders, can modify the header "User-Agent" from customResponseHeaders. I want to know, traefik whether 
prohibit modification of host header ？

## This is the header yaml
```
apiVersion: traefik.containo.us/v1alpha1
kind: Middleware
metadata:
  name: add-header
spec:
  headers:
    customRequestHeaders:
      Host: "dev-cn-mercku-static-files.s3-website.cn-north-1.amazonaws.com.cn"
      X-Forwarded-Host: "dev-cn-mercku-static-files.s3-website.cn-north-1.amazonaws.com.cn"
    customResponseHeaders:
      User-Agent: "bbbbbb"
```

## This is the curl result
```
sifksky@sifksky-Office:~$ curl -vvv http://kk.hyku.org
* Rebuilt URL to: http://kk.hyku.org/
*   Trying 10.70.1.34...
* TCP_NODELAY set
* Connected to kk.hyku.org (10.70.1.34) port 80 (#0)
> GET / HTTP/1.1
> Host: kk.hyku.org
> User-Agent: curl/7.58.0
> Accept: */*
> 
< HTTP/1.1 200 OK
< Content-Length: 487
< Content-Type: text/plain; charset=utf-8
< Date: Wed, 11 Dec 2019 09:27:32 GMT
< User-Agent: bbbbbb
< 
Hostname: whoami-8457d7cd94-872cp
IP: 127.0.0.1
IP: 10.244.0.73
RemoteAddr: 10.244.0.94:54382
GET / HTTP/1.1
Host: kk.hyku.org
User-Agent: curl/7.58.0
Accept: */*
Accept-Encoding: gzip
X-Forwarded-For: 10.70.1.5
X-Forwarded-Host: dev-cn-mercku-static-files.s3-website.cn-north-1.amazonaws.com.cn
X-Forwarded-Port: 80
X-Forwarded-Proto: http
X-Forwarded-Server: traefik-xbqwk
X-Instana-L: 1
X-Instana-S: 59d830a6c704be06
X-Instana-T: 5015f4c577283c64
X-Real-Ip: 10.70.1.5

* Connection #0 to host kk.hyku.org left intact
```

**Comment by traefiker** [comment]
Hi! I'm Træfiker :robot: the bot in charge of communication regulation.

Thanks for your interest in Traefik!

We dedicate the issue tracker to bug reports and feature requests only. My advanced AI has spotted that your issue might be a configuration problem or relates to something that doesn't look like a bug.

To confirm this, please join our [Community Forum](https://community.containo.us/) and reach out to us on the [Traefik section](https://community.containo.us/c/traefik).

In case I'm wrong (well, that would be embarrassing :sweat_smile:), my developers will re-open the issue and fix me!

In the meantime, you can double check [Traefik's documentation](https://docs.traefik.io/).

**Comment by marcandrews** [comment]
@ldez, I was wondering if we could revisit this issue.

I am trying to use Traefik to serve public static assets from Google Cloud Storage. Here is my configuration:
```yaml
kind: Service
apiVersion: v1
metadata:
  name: web-master-spa
  namespace: default
spec:
  type: ExternalName
  externalName: storage.googleapis.com
  ports:
    - port: 80
---
apiVersion: traefik.containo.us/v1alpha1
kind: Middleware
metadata:
  name: web-middleware-headers
spec:
  headers:
    customRequestHeaders:
      Host: 'example.com'
---
apiVersion: traefik.containo.us/v1alpha1
kind: Middleware
metadata:
  name: web-middleware-prefix
spec:
  addPrefix:
    prefix: /master/spa
---
apiVersion: traefik.containo.us/v1alpha1
kind: IngressRoute
metadata:
  name: web-ingress-routes
  namespace: default
spec:
  entryPoints:
    - web
  routes:
    - kind: Rule
      match: PathPrefix(`/`)
      services:
        - kind: Service
          name: web-master-spa
          port: 80
      middlewares:
        - name: web-middleware-headers
        - name: web-middleware-prefix

```

The full URL to my bucket is `http://storage.googleapis.com/example.com/master/spa/index.htm`.

In order to serve this object from `example.com/index.htm`, the `Host` header either:
1. needs to be set to `example.com`, making the object accessible from `http://storage.googleapis.com/master/spa/index.htm` (Google uses the `Host` header to determine the bucket name) with the `/master/spa` `addPrefix` middleware; or
2. needs to be set to `storage.googleapis.com`, making the object accessible from the full URL with `example.com/master/spa` `addPrefix` middleware.

This will work when deployed to `example.com`, however, as soon as it is deployed somewhere else, like `localhost`, or I need to proxy to another bucket, it breaks.

I read on the [forum](https://community.containo.us/t/forward-host-header/1065/2) from @dtomcej that the `Host` header is critical to how the proxy works with dynamic configurations, but is there anyway to allow the `Host` header to be modified at least for `ExternalName` services?

Related resources:
https://community.containo.us/search?q=host%20header%20category%3A10
https://community.containo.us/t/traefik-can-not-modify-host-header/3358
https://community.containo.us/t/forward-host-header/1065
https://community.containo.us/t/trying-to-understand-traefik-host-header/850

**Comment by traefiker** [comment]
Closed by #6502.
